### PR TITLE
[DP-35]

### PR DESCRIPTION
**Pull request by Audiosutras**
This PR adds the following to `map_petitioner`, `map_attorneys`, and `map_agencies`:
https://caktus.atlassian.net/browse/DP-35

Added a migration and custom migration for supporting `state` choices on the `contact.state` field and `GeneratePetitionSerializer`'s `drivers_license_state` field.

### CODE REVIEW

File: dear_petition/petition/api/serializers.py [hunk 1]
```diff
@@ -89,6 +90,8 @@ class GeneratePetitionSerializer(serializers.Serializer):
     )
     ssn = serializers.CharField(label="SSN")
     drivers_license = serializers.CharField(label="Driver's License #")
+    drivers_license_state = serializers.ChoiceField(choices=us_states.US_STATES)
+    # drivers_license_state = serializers.CharField(label="State of Driver's License")
```
Comment (copelco) [inline dear_petition/petition/api/serializers.py:94]:
```suggestion
```

File: dear_petition/petition/conftest.py [hunk 2]
```diff
@@ -15,6 +16,11 @@ def data():
     return {}
 
 
+@pytest.fixture
+def extra():
+    return {}
```
Comment (copelco) [inline dear_petition/petition/conftest.py:21]: 🆒

File: dear_petition/petition/export/mapper.py [hunk 3]
```diff
@@ -18,12 +19,13 @@ def build_pdf_template_context(petition, extra):
 
 
 def map_petition(data, petition, extra={}):
-    data["County"] = petition.county
-    if petition.jurisdiction == constants.DISTRICT_COURT:
+    data["County"] = getattr(petition, "county", None)
```
Comment (copelco) [inline dear_petition/petition/export/mapper.py:22]: The model requires these fields, so they should always be set. I think we can just use the ``data["County"] = petition.county`` format if possible.

File: dear_petition/petition/export/mapper.py [hunk 4]
```diff
@@ -34,6 +36,9 @@ def map_petitioner(data, petition, extra={}):
     data["NamePetitioner"] = getattr(
         record, "label", None
     )  # load.py line 28 (label is set to name attr)
+    data["SSN"] = extra.get("ssn", None)
```
Comment (copelco) [inline dear_petition/petition/export/mapper.py:39]: The serializer requires these fields. Did we run into issues with these fields not being set?

File: dear_petition/petition/export/mapper.py [hunk 5]
```diff
@@ -44,157 +49,34 @@ def map_petitioner(data, petition, extra={}):
 
 
 def map_attorney(data, petition, extra={}):
-    pass
+    attorney = extra.get("attorney", None)
+    data["NameAtty"] = getattr(attorney, "name", None)
```
Comment (copelco) [inline dear_petition/petition/export/mapper.py:53]: Attorney attributes are allowed to be ``blank`` on the model, so I think the default empty value of an empty string should be fine for these, e.g. ``attorney.address1``.

File: dear_petition/petition/export/mapper.py [hunk 6]
```diff
@@ -44,157 +49,34 @@ def map_petitioner(data, petition, extra={}):
...
+    data["PetitionNotFiledSignName"] = getattr(attorney, "name", None)
+    data["PetitionerAttorneyCbx"] = Checkbox("Yes")
+    data["PetitionNotFiledSignDate"] = dt_obj_to_date(dt.datetime.today()).strftime(
+        constants.DATE_FORMAT
+    )
 
 
 def map_agencies(data, petition, extra={}):
-    pass
+    agencies = extra.get("agencies", {})
```
Comment (copelco) [inline dear_petition/petition/export/mapper.py:70]: I think agencies is guaranteed to be a list from the serializer, so I think we can just iterate over these, e.g.:

```python
for idx, agency in enumerate(agencies):
    data[f"NameAgency{idx}"] = agency.name
    # ...
```

File: dear_petition/petition/export/mapper.py [hunk 7]
```diff
@@ -44,157 +49,34 @@ def map_petitioner(data, petition, extra={}):
...
+        data["CityAgency1"] = getattr(agency1, "city", None)
+        data["StateAgency1"] = getattr(agency1, "state", None)
+        data["ZipAgency1"] = getattr(agency1, "zipcode", None)
 
 
 def map_offenses(data, petition, extra={}):
     pass
-
-
-def map_data(form_data, batch):
```
Comment (copelco) [inline dear_petition/petition/export/mapper.py:58]: 💥

File: dear_petition/petition/export/tests/test_mapper.py [hunk 8]
```diff
@@ -67,3 +69,115 @@ def test_map_petitioner__dob(data, petition, record1):
...
+def test_map_agencies__mail_address(data, petition, extra, contact):
+    extra["agencies"] = [contact]
+    mapper.map_agencies(data, petition, extra)
+    assert data["MailAgency1"] == contact.address2
+
+
+def test_map_agencies__city(data, petition, extra, contact):
+    extra["agencies"] = [contact]
+    mapper.map_agencies(data, petition, extra)
+    assert data["CityAgency1"] == contact.city
```
Comment (copelco) [inline dear_petition/petition/export/tests/test_mapper.py:171]: nice tests!

File: dear_petition/petition/models.py [hunk 9]
```diff
@@ -237,7 +238,7 @@ class Contact(models.Model):
     address1 = models.CharField("Address (Line 1)", max_length=512, blank=True)
     address2 = models.CharField("Address (Line 2)", max_length=512, blank=True)
     city = models.CharField(max_length=64, blank=True)
-    state = models.CharField(max_length=64, blank=True)
+    state = models.CharField(choices=us_states.US_STATES, max_length=64, blank=True)
```
Comment (copelco) [inline dear_petition/petition/models.py:241]: 💯

File: dear_petition/petition/export/mapper.py (hunk 4 above)
Comment (Audiosutras) [reply dear_petition/petition/export/mapper.py:38]: I think it's an issue with `pdf writer`. I am able to see `data["SSN"]` and `extra["ssn"]`.

File: dear_petition/petition/export/mapper.py (hunk 5 above)
Comment (Audiosutras) [reply dear_petition/petition/export/mapper.py:52]: done

File: dear_petition/petition/export/mapper.py (hunk 6 above)
Comment (Audiosutras) [reply dear_petition/petition/export/mapper.py:69]: done and updated test

File: dear_petition/petition/export/mapper.py (hunk 7 above)
Comment (Audiosutras) [reply dear_petition/petition/export/mapper.py:58]: 🎆

File: dear_petition/petition/export/mapper.py (hunk 4 above)
Comment (copelco) [reply dear_petition/petition/export/mapper.py:38]: Ah, it's because the PDF has a bad key for it, ``SNN``. It should work with that.

**Comment by copelco** [comment]
Looks good after fixing SNN

**Comment by Audiosutras** [comment]
> Looks good after fixing SNN

I made one more change. its to `map_agencies`. I think we want to start the for loop index at 1.

File: dear_petition/petition/export/mapper.py [hunk 10]
```diff
@@ -34,6 +35,9 @@ def map_petitioner(data, petition, extra={}):
     data["NamePetitioner"] = getattr(
         record, "label", None
     )  # load.py line 28 (label is set to name attr)
+    data["SSN"] = extra.get("ssn", None)
```
Comment (copelco) [inline dear_petition/petition/export/mapper.py:38]:
```suggestion
    # note: SNN and note SSN due to bug in PDF field name
    data["SNN"] = extra.get("ssn", None)
```

**Pull request merged**